File: app/src/main/java/com/roman/batsu/ui/home/HomeContract.java
package com.roman.batsu.ui.home;

import com.roman.batsu.ui.home.pojo.Home;

import java.util.List;

public interface HomeContract {

    //то что происходит в активити / fragments
    interface View {
        void showProgress();
        void hideProgress();
        void setDataToRecyclerView(List<Home> movieArrayList);
        void onResponseFailure(Throwable throwable);
    }

    //то, что происходит в презентере
    interface Presenter {
        void onDestroy();
        void requestDataFromServer(String filename);
    }



    interface Model {
        interface OnFinishedListener {
            void onFinished(List<Home> homeList);
            void onFailure(Throwable t);
        }

        void getMovieList(OnFinishedListener onFinishedListener, String fileName);

    }

}

File: app/src/main/java/com/roman/batsu/ui/main/HomeContract.kt
package com.roman.batsu.ui.main

import com.roman.batsu.ui.model.Home

interface HomeContract {
    //то что происходит в активити / fragments
    interface View {
        fun showProgress()
        fun hideProgress()
        fun setDataToRecyclerView(movieArrayList: MutableList<Home>)
        fun onResponseFailure(throwable: Throwable)
    }

    //то, что происходит в презентере
    interface Presenter {
        fun onDestroy()
        fun requestDataFromServer(filename: String)
    }

    interface Model {
        interface OnFinishedListener {
            fun onFinished(homeList: MutableList<Home>)
            fun onFailure(t: Throwable)
        }

        fun getMovieList(onFinishedListener: OnFinishedListener, fileName: String)
    }
}